tests/vmi_cloudinit_test.go

```diff
@@ -258,54 +260,38 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 			})
 		})
 
-		Context("with cloudInitNoCloud userData source", func() {
-			It("[test_id:1617]should process provided cloud-init data", func() {
-				userData := fmt.Sprintf("#!/bin/sh\n\necho '%s'\n", expectedUserData)
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), userData)
-
-				vmi = LaunchVMI(vmi)
-				CheckCloudInitIsoSize(vmi, cloudinit.DataSourceNoCloud)
-				By("executing a user-data script")
-				VerifyUserDataVMI(vmi, []expect.Batcher{
-					&expect.BSnd{S: "\n"},
-					&expect.BExp{R: expectedUserData},
-				}, time.Second*120)
-
-				By("applying the hostname from meta-data")
-				Expect(console.LoginToCirros(vmi)).To(Succeed())
-
-				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
-					&expect.BSnd{S: "hostname\n"},
-					&expect.BExp{R: dns.SanitizeHostname(vmi)},
-				}, 10)).To(Succeed())
-			})
-		})
-
-		Context("with cloudInitConfigDrive userData source", func() {
-			It("[test_id:3180]should process provided cloud-init data", func() {
-				userData := fmt.Sprintf("#!/bin/sh\n\necho '%s'\n", expectedUserData)
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndConfigDriveUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), userData)
-
-				vmi = LaunchVMI(vmi)
-				CheckCloudInitIsoSize(vmi, cloudinit.DataSourceConfigDrive)
-				By("executing a user-data script")
-				VerifyUserDataVMI(vmi, []expect.Batcher{
-					&expect.BSnd{S: "\n"},
-					&expect.BExp{R: expectedUserData},
-				}, time.Second*120)
-
-				By("applying the hostname from meta-data")
-				Expect(console.LoginToCirros(vmi)).To(Succeed())
-
-				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
-					&expect.BSnd{S: "hostname\n"},
-					&expect.BExp{R: dns.SanitizeHostname(vmi)},
-				}, 10)).To(Succeed())
-			})
-		})
+		DescribeTable("should process provided cloud-init data", func(vmi *v1.VirtualMachineInstance, dsType cloudinit.DataSourceType) {
+			vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
+
+			By("waiting until login appears")
+			libwait.WaitUntilVMIReady(vmi, console.LoginToCirros)
+
+			By("validating cloud-init disk is 4k aligned")
+			CheckCloudInitIsoSize(vmi, dsType)
+
+			By("Checking whether the user-data script had created the file")
+			Expect(console.RunCommand(vmi, fmt.Sprintf("cat /%s\n", expectedUserDataFile), time.Second*120)).To(Succeed())
+
+			By("validating the hostname matches meta-data")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "hostname\n"},
+				&expect.BExp{R: dns.SanitizeHostname(vmi)},
+			}, 10)).To(Succeed())
+		},
+			Entry("[test_id:1617] with cloudInitNoCloud userData source",
+				tests.NewRandomVMIWithEphemeralDiskAndUserdata(
+					cd.ContainerDiskFor(cd.ContainerDiskCirros),
+					fmt.Sprintf("#!/bin/sh\n\ntouch /%s\n", expectedUserDataFile)),
+				cloudinit.DataSourceNoCloud),
```

Comment:
This cannot be used directly. `tests.NewRandomVMIWithEphemeralDiskAndUserdata` uses `NewRandomVMI` which at the end makes an Expectation and this cannot be done in container nodes https://onsi.github.io/ginkgo/#no-assertions-in-container-nodes
This is why this failure:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/9242/pull-kubevirt-check-unassigned-tests/1628028781794955264
:)

Reply:
@fossedihelm I tested it locally with `FDescribetable..` that's why I didn't see this error. Will fix it soon